src/tracing/agent.cc

```diff
@@ -184,5 +187,38 @@ void Agent::Flush(bool blocking) {
   for (const auto& id_writer : writers_)
     id_writer.second->Flush(blocking);
 }
+
+ClientHandle::~ClientHandle() {
+  agent_->Disconnect(id_);
+}
+
+AgentHandle::AgentHandle(Agent* agent) : agent_(agent) {}
+
+void AgentHandle::Reset() {
+  Mutex::ScopedLock scoped_lock(mutex_);
+  agent_ = nullptr;
+}
+
+void AgentHandle::AppendTraceEvent(TraceObject* trace_event) {
+  Mutex::ScopedLock scoped_lock(mutex_);
+  if (agent_ != nullptr) {
+    agent_->AppendTraceEvent(trace_event);
+  }
+}
+
+void AgentHandle::Flush(bool blocking) {
+  Mutex::ScopedLock scoped_lock(mutex_);
+  if (agent_ != nullptr) {
+    agent_->Flush(blocking);
+  }
+}
+
+void AgentHandle::Disconnect(int client) {
+  // This is supposed to be only callable from a main thread, same as Reset
+  // This code needs to be updated to enforce that.
+  if (agent_ != nullptr)
```

Comment:
Shouldn't this method take out the lock first?

Reply:
I cannot add lock without significant rework (which would delay this PR) because node::Mutex is not reentrant... Currently this is only called on the isolate thread (i.e. it will be be correct for webworkers) so it is correct without the lock... But it is not obvious

Reply:
Okay, can you update the comment and maybe make a bit more imperative?  "Is supposed to" isn't strong enough.

Reply:
Done.